File: rna/rna.rb
input = File.open(ARGV[0].to_s, "r")
output = File.open(ARGV[1].to_s, "w")

def change_amino_acid(codon)

end

aa_unique = [:Met, :Trp]

aa_dictionary = {:Phe => ["UUU", "UUC"], :Leu => ["UUA", "UUG", "CUU", "CUC", "CUA", "CUG"],
	:Ser => ["UCU", "UCC", "UCA", "UCG"], :Stop => ["UAA", "UAG", "UGA"], :Arg => ["CGG", "CGA", "CGC", "CGU", "AGG", "AGA"]}


n = input.gets.to_i # number of test cases
n.times do |variable|
	rna_input = input.gets
	codons = []
	(rna_input.length / 3).times do |i|
		puts rna_input.slice(i*3, 3)
		codons << rna_input.slice(i*3, 3)
	end
	codons = rna_input.split()
	n_intervals = input.gets.to_i
	intervals = []
	n_intervals.times do
		intervals << input.gets.split(" ")

		# xsite = rsites[0]
		# ysite = rsites[1]
		# if ysite <= 3
		# 	output.puts("-1")
		# end
	end
	puts intervals
	puts codons
end

File: rna/rna.py
import sys
import numpy as np

f = open(sys.argv[1], 'r')
g = open(sys.argv[2], 'w')


def change_amino_acid(codon):
	if codon[0] == "U":
		if codon[1] == "U":
			if codon[2] == "U":
				return "C"

class AminoAcid(str):

	def identify(codon):
		if 

	phenylalanine = ["UUU", "UUC"]
	leucine = ["UUA", "UUG", "CUU", "CUC", "CUA", "CUG"]

	class isoleucine

	class methionine

	class valine

	class serine

	class proline

	class threonine

	class alanine

	class tyrosine

	class stop

	class histidine

	class glutamine

	class asparagine

	class lysine

	class aspartate

	class glutamate

	class cysteine

	class tryptophan

	class arginine

	class glycine


n = int(f.readline()) # number of test cases
while n > 0 :
	rna_input = f.readline()
	n_intervals = f.readline()
	while n_intervals > 0 :
		rsites = f.readline().split(" ")
		xsite = rsites[0]
		ysite = rsites[1]
		if ysite >= 3 :
			g.writeline("-1") # Not possible to change start codon
		n_intervals -= 1